OpenRA.Mods.Common/Traits/Conditions/ToggleChargedConditionOnOrder.cs

```diff
@@ -150,8 +153,13 @@ Order IIssueDeployOrder.IssueDeployOrder(Actor self, bool queued)
 
 		public void ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString == "DeployWithCharge")
-				self.QueueActivity(order.Queued, new DeployForGrantedConditionWithCharge(self, this));
+			if (order.OrderString != "DeployWithCharge")
+				return;
+
+			if (order.Queued || Info.CancelsCurrentOrders)
+				self.QueueActivity(order.Queued, new ToggleChargedCondition(self, this));
+			else if (CanDeploy())
+				TriggerDeploy(self);
```

Comment:
```suggestion
				ToggleDeploy(self);
```
(or ToggleState or something since @pchote wanted to remove the "deploy" terminology)